src/middlewared/middlewared/plugins/sysdataset.py

```diff
@@ -288,22 +288,23 @@ async def setup(self, exclude_pool):
                                              {'extra': {'retrieve_children': False}})
         if not dataset or dataset[0]['locked']:
             # Pool is not mounted (e.g. HA node B), temporary set up system dataset on the boot pool
-            self.middleware.logger.debug(
+            self.logger.debug(
                 'Root dataset for pool %r is not available, temporarily setting up system dataset on boot pool',
                 config['pool'],
             )
             self.force_pool = boot_pool
             config = await self.config()
 
-        mounted_pool = None
+        mounted_pool = mounted = None
         for p in psutil.disk_partitions():
             if p.mountpoint == SYSDATASET_PATH:
                 mounted_pool = p.device.split('/')[0]
         if mounted_pool and mounted_pool != config['pool']:
-            self.middleware.logger.debug('Abandoning dataset on %r in favor of %r', mounted_pool, config['pool'])
+            self.logger.debug('Moving system dataset from %r to %r', mounted_pool, config['pool'])
             async with self._release_system_dataset():
                 await self.__umount(mounted_pool, config['uuid'])
                 await self.__setup_datasets(config['pool'], config['uuid'])
+                mounted = await self.__mount(config['pool'], config['uuid'])
```

Comment:
Don't we need to do the `makedirs`/ACL stuff before mounting?

Reply:
We just need to make sure the acls are turned `off` for the dataset. This made me look at that code and changing the ACL logic is actually incorrect so that's another bug I need to fix.